gating/update_dependencies/run

```diff
@@ -1,11 +1,47 @@
-#!/bin/bash -xe
+#!/bin/bash
 
+# Set options to ensure easier debugging
+# and failure on any problems encountered
+set -euxo pipefail
+
+# Find the absolute path to the directory this script
+# is executed from.
 export GATING_PATH="$(readlink -f $(dirname ${0}))"
-source "${GATING_PATH}/../../scripts/functions.sh"
+
+# Set the base RPC-O directory which the functions use
+export BASE_DIR="$(readlink -f ${GATING_PATH}/../..)"
+
+# As get-rpc_release.py requires a yaml import, make
+# sure that it is installed and available.
+if ! python -c 'import yaml' &>/dev/null; then
+  echo "Python YAML library not available. Installing it."
+  apt-get install -y python-yaml
+fi
+
+# Source the functions
+source "${BASE_DIR}/scripts/functions.sh"
+
+# Validate that RPC_RELEASE is set and has a value
```

Comment:
Is there are reason this only needs to be validated for the dependency update? Maybe it would be better to have this validation in the functions file. I think other scripts should be able to expect that if sourcing the file didn't fail, things are set up correctly. Validating in that file also means that all jobs/deployments benefit from this check.

Reply:
Good call - moved this to the functions.